report/report-generator.js

```diff
@@ -102,13 +93,58 @@ class ReportGenerator {
     return '<script src="/pages/scripts/lighthouse-report.js"></script>';
   }
 
+  _createSections(aggregations) {
+    return aggregations.reduce((menu, aggregation) => {
+      if (menu.indexOf(aggregation.type.name) === -1) {
+        menu.push(aggregation.type.name);
+      }
+      return menu;
+    }, []);
+  }
+
+  _createPWAAuditsByTag(aggregations) {
+    const items = {};
+
+    aggregations.forEach(aggregation => {
+      // We only regroup the PWA aggregations around so ignore
+      if (aggregation.type.name !== Aggregate.VALID_TYPES.PWA.name) {
+        return;
+      }
+
+      aggregation.score.subItems.forEach(subItem => {
+        // Create a space for the tag.
+        if (!items[subItem.tag]) {
+          items[subItem.tag] = {};
+        }
+
+        // Then use the name to de-dupe the same audit from different aggregations.
+        if (!items[subItem.tag][subItem.name]) {
+          items[subItem.tag][subItem.name] = subItem;
+        }
+      });
+    });
+
+    return items;
+  }
+
   generateHTML(results, options) {
     const inline = (options && options.inline) || false;
+    const totalAggregations = results.aggregations.reduce((total, aggregation) => {
+      if (!aggregation.type.contributesToScore) {
```

Comment:
This is just to account for Best Practices, which don't affect the overall scoring.